Fetch more details from the orders api

knewter opened the pull request:
There were some Mt. Gox Order Details that weren't being fetched into
the local MtGox::Order object.  This change fetches those and makes them
accessible, as well as preferring BigDecimal to Float.

What are your thoughts on these changes @sferik?  I was mildly concerned when I saw floats being used, and so that was an unexpected change I introduced.  The other is really just proxying data through that they provide that we weren't providing.

Commit:
Fetch more details from the orders api

There were some Mt. Gox Order Details that weren't being fetched into
the local MtGox::Order object.  This change fetches those and makes them
accessible, as well as preferring BigDecimal to Float.

sferik commented on .rvmrc:
```diff
@@ -0,0 +1 @@
+rvm 1.9.3@mtgox --create
```
This file probably doesn’t belong in the repo.

knewter commented:
Sorry about that @sferik.  I force-pushed without .rvmrc

sferik commented:
I'd be ready to merge this patch if it didn't change `Float` to `BigDecimal`. I understand the reasons for preferring `BigDecimal` to `Float`, but there are various other places in this library where floats are used and I think it's important to be consistent. Would you mind changing the `BigDecimal` back to `Float` or doing this conversion everywhere in the library?

Commit:
Convert all floats to BigDecimal

The changes made to values in the tests are actually corrections from
previously inaccurate calculations due to floating point storage.  They
would have been potentially inaccurate on different architectures, and
did not represent the actual underlying math.

knewter commented:
@sferik so I went through and converted everything to BigDecimal.  Anything else you'd like to see in this?

sferik commented:
It looks like your latest changes broke the build.

knewter commented:
@sferik it works fine for me, and that change should not have dropped coverage.  Are you saying that tests are failing, or that coveralls reports too little coverage?  I even tried fetching upstream, merging master, to see if I could cause a test to fail, but I just get this:

--($:~/ruby/mtgox)(feature/grab_more_order_data)ruby-1.9.3-p392@mtgox-- 
$ rake
/media/jadams/ssd/home/jadams/.rvm/rubies/ruby-1.9.3-p392/bin/ruby -S rspec ./spec/faraday/response_spec.rb ./spec/mtgox/client_spec.rb ./spec/mtgox_spec.rb
/home/jadams/.rvm/gems/ruby-1.9.3-p392@mtgox/gems/debugger-1.5.0/lib/ruby-debug/printers/base.rb:7: warning: already initialized constant SEPARATOR
................................

Finished in 0.36638 seconds
32 examples, 0 failures

Randomized with seed 3090

Coverage report generated for RSpec to /home/jadams/ruby/mtgox/coverage. 561 / 565 LOC (99.29%) covered.
[Coveralls] Outside the Travis environment, not sending data.

sferik commented on lib/mtgox/lag.rb:
```diff
@@ -1,10 +1,11 @@
+require 'pry'
```
I believe the build failures are being caused by this line. See https://travis-ci.org/sferik/mtgox/builds/6957565 for details.

knewter commented:
Urk, I'm on it.  Sorry.

![Facepalm](http://i0.kym-cdn.com/photos/images/original/000/001/582/picard-facepalm.jpg?1240934151)

knewter commented:
So I force pushed, it should be better.  I'll go check travis.

sferik commented:
Looks like tests are still failing on 1.9.2. https://travis-ci.org/sferik/mtgox/builds/6966514

Commit:
Coerce inputs to BigDecimal to_s

In ruby 1.9.2, BigDecimal didn't handle being passed a non-string for
its convenience function.  This makes the BigDecimal change support
1.9.2

knewter commented:
@sferik bah, forgot that 1.9.2 did that with BigDecimal.  I've pushed a fix, should fix it for reals finally.  Once again, Travis saved the day :)

sferik commented:
My preference is for `require 'bigdecimal'` to appear at the top of each individual class that requires it, instead of once in `lib/mtgox.rb`. This way classes can be required or tested independent of the entire library. Other than that, this patch is looking good.

knewter commented:
Urk. That's what I usually do but I've had people complain and didn't see
it in your code base. Thanks
On May 9, 2013 7:48 AM, "Erik Michaels-Ober" notifications@github.com
wrote:

> I'm prefer you require 'bigdecimal' appeared at the top of each
> individual class that requires it instead of once in lib/mtgox.rb. This
> way classes can be required or tested independent of the entire library.
> Other than that, this patch is looking good.
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/sferik/mtgox/pull/42#issuecomment-17662511
> .

Commit:
Require 'bigdecimal' only where necessary

knewter commented:
How's that looking now @sferik?

sferik commented:
@knewter I think you may have missed a spot in `lib/mtgox/trade.rb`.

knewter commented:
```
class Trade < Order
```

Then https://github.com/sferik/mtgox/pull/42/files#L3R2

Essentially, it'd be redundant since Trade already depends on/derives from order, and order brings it in.  I can add it if you want but it seems extraneous at that point.

sferik commented:
Nope, that makes sense. Thanks for submitting this pull request.
